client.go

```diff
@@ -114,7 +127,7 @@ func (c *Client) httpRequest(
 }
 
 func (c *Client) requestPath(path string) string {
-	return fmt.Sprintf("%s:%v/%s", c.hostname, c.port, path)
+	return fmt.Sprintf("http://%s:%v/%s", c.hostname, c.port, path)
```

Comment:
Does this reduce our ability to support HTTPS deployments later?

Reply:
In retrospect, yes it does. 
I changed it because it was storing a URL in the variable "hostname". Also the "URL" had the port separated in a separate variable.
How about a better fix, where we drop the "hostname" and "port" variable and just have URL??